src/uucore/src/lib/mods/backup_control.rs

```diff
@@ -17,15 +101,151 @@ the VERSION_CONTROL environment variable.  Here are the values:
   existing, nil   numbered if numbered backups exist, simple otherwise
   simple, never   always make simple backups";
 
+static VALID_ARGS_HELP: &str = "Valid arguments are:
+  - ‘none’, ‘off’
+  - ‘simple’, ‘never’
+  - ‘existing’, ‘nil’
+  - ‘numbered’, ‘t’";
+
+/// Available backup modes.
+///
+/// The mapping of the backup modes to the CLI arguments is annotated on the
+/// enum variants.
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub enum BackupMode {
+    /// Argument 'none', 'off'
     NoBackup,
+    /// Argument 'simple', 'never'
     SimpleBackup,
+    /// Argument 'numbered', 't'
     NumberedBackup,
+    /// Argument 'existing', 'nil'
     ExistingBackup,
 }
 
-pub fn determine_backup_suffix(supplied_suffix: Option<&str>) -> String {
+/// Backup error types.
+///
+/// Errors are currently raised by [`determine_backup_mode`] only. All errors
+/// are implemented as [`UCustomError`] for uniform handling across utilities.
+#[derive(Debug, Eq, PartialEq)]
+pub enum BackupError {
+    /// An invalid argument (e.g. 'foo') was given as backup type. First
+    /// parameter is the argument, second is the arguments origin (CLI or
+    /// ENV-var)
+    InvalidArgument(String, String),
+    /// An ambiguous argument (e.g. 'n') was given as backup type. First
+    /// parameter is the argument, second is the arguments origin (CLI or
+    /// ENV-var)
+    AmbiguousArgument(String, String),
+    /// Currently unused
+    BackupImpossible(),
+    // BackupFailed(PathBuf, PathBuf, std::io::Error),
+}
+
+impl UError for BackupError {
+    fn code(&self) -> i32 {
+        match self {
+            BackupError::BackupImpossible() => 2,
+            _ => 1,
+        }
+    }
+
+    fn usage(&self) -> bool {
+        // Suggested by clippy.
+        matches!(
+            self,
+            BackupError::InvalidArgument(_, _) | BackupError::AmbiguousArgument(_, _)
+        )
+    }
+}
+
+impl Error for BackupError {}
+
+impl Display for BackupError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use BackupError as BE;
+        match self {
+            BE::InvalidArgument(arg, origin) => write!(
+                f,
+                "invalid argument ‘{}’ for ‘{}’\n{}",
```

Comment:
Quotes are dependent on locale, so while we don't have proper locale support, please use the C locale (which uses `'`). You can run GNU utils with `LC_ALL=C [utility] [args]` to see what the C locale output is.